bin/setenv.sh

```diff
@@ -0,0 +1,179 @@
+xmlstarlet ed -L -d '//Connector[@protocol="AJP/1.3"]' ${INSTALL_DIR}/conf/server.xml || {
+  echo "ERROR during xmlstarlet processing (AJP connector removal)"
+  exit 1
+}
+
+# Tomcat HTTP Thread pool configuration
+xmlstarlet ed -L -s "/Server/Service/Connector" -t attr -n "maxThreads" -v "${HTTP_THREAD_MAX}" \
+  -s "/Server/Service/Connector" -t attr -n "minSpareThreads" -v "${HTTP_THREAD_MIN}" \
+  ${INSTALL_DIR}/conf/server.xml || {
+  echo "ERROR during xmlstarlet processing (configuring threads)"
+  exit 1
+}
+
+# Proxy configuration
+xmlstarlet ed -L -s "/Server/Service/Connector" -t attr -n "proxyName" -v "${PROXY_VHOST}" ${INSTALL_DIR}/conf/server.xml || {
+  echo "ERROR during xmlstarlet processing (adding Connector proxyName)"
+  exit 1
+}
+
+if [ "${PROXY_SSL}" = "true" ]; then
+  xmlstarlet ed -L -s "/Server/Service/Connector" -t attr -n "scheme" -v "https" \
+    -s "/Server/Service/Connector" -t attr -n "secure" -v "false" \
+    -s "/Server/Service/Connector" -t attr -n "proxyPort" -v "${PROXY_PORT}" \
+    ${INSTALL_DIR}/conf/server.xml || {
+    echo "ERROR during xmlstarlet processing (configuring Connector proxy ssl)"
+    exit 1
+  }
+else
+  xmlstarlet ed -L -s "/Server/Service/Connector" -t attr -n "scheme" -v "http" \
+    -s "/Server/Service/Connector" -t attr -n "secure" -v "false" \
+    -s "/Server/Service/Connector" -t attr -n "proxyPort" -v "${PROXY_PORT}" \
+    ${INSTALL_DIR}/conf/server.xml || {
+    echo "ERROR during xmlstarlet processing (configuring Connector proxy)"
+    exit 1
+  }
+fi
+
+# Add a new valve to replace the proxy ip by the client ip (just before the end of Host)
+xmlstarlet ed -L -s "/Server/Service/Engine/Host" -t elem -n "ValveTMP" -v "" \
+  -i "//ValveTMP" -t attr -n "className" -v "org.apache.catalina.valves.RemoteIpValve" \
+  -i "//ValveTMP" -t attr -n "remoteIpHeader" -v "x-forwarded-for" \
+  -i "//ValveTMP" -t attr -n "proxiesHeader" -v "x-forwarded-by" \
+  -i "//ValveTMP" -t attr -n "protocolHeader" -v "x-forwarded-proto" \
+  -r "//ValveTMP" -v Valve \
+  ${INSTALL_DIR}/conf/server.xml || {
+  echo "ERROR during xmlstarlet processing (adding RemoteIpValve)"
+  exit 1
+}
+
+# JMX configuration
+if [ "${JMX_ENABLED}" = "true" ]; then
+  # insert the listener before the "Global JNDI resources" line
+  xmlstarlet ed -L -i "/Server/GlobalNamingResources" -t elem -n ListenerTMP -v "" \
+    -i "//ListenerTMP" -t attr -n "className" -v "org.apache.catalina.mbeans.JmxRemoteLifecycleListener" \
+    -i "//ListenerTMP" -t attr -n "rmiRegistryPortPlatform" -v "${JMX_RMI_REGISTRY_PORT}" \
+    -i "//ListenerTMP" -t attr -n "rmiServerPortPlatform" -v "${JMX_RMI_SERVER_PORT}" \
+    -i "//ListenerTMP" -t attr -n "useLocalPorts" -v "false" \
+    -r "//ListenerTMP" -v "Listener" \
+    ${INSTALL_DIR}/conf/server.xml || {
+    echo "ERROR during xmlstarlet processing (adding JmxRemoteLifecycleListener)"
+    exit 1
+  }
+
+  CATALINA_OPTS="${CATALINA_OPTS} -Dcom.sun.management.jmxremote=true"
```

Comment:
The first `CATALINA_OPTS` variable usage could raise an error due to the script settings (`#!/bin/bash -eu`)

```suggestion
  CATALINA_OPTS="${CATALINA_OPTS:-} -Dcom.sun.management.jmxremote=true"
```